python/lib/evaluations/dmod/evaluations/specification/base.py

```diff
@@ -456,28 +466,51 @@ def create(
         return instance
 
     @abc.abstractmethod
-    def validate(self) -> typing.Sequence[str]:
+    def validate_self(self) -> typing.Sequence[str]:
         """
         Returns:
             Any messages indicating a problem with the specification
         """
         pass
 
-    @abc.abstractmethod
-    def extract_fields(self) -> typing.Dict[str, typing.Any]:
+    def extract_fields(
+        self,
+        *,
+        include: typing.Optional[typing.Union[typing.Set, typing.Mapping]] = None,
+        exclude: typing.Optional[typing.Union[typing.Set, typing.Mapping]] = None,
+        by_alias: bool = None,
+        exclude_unset: bool = None,
+        exclude_defaults: bool = None,
+        exclude_none: bool = None
+    ) -> typing.Dict[str, typing.Any]:
         """
         Returns:
             Specification specific fields that will fit within a final serialized representation
         """
-        fields = {
-            "properties": self.properties.copy() if self.properties else dict()
-        }
+        if exclude_defaults is None:
```

Comment:
It's a basic protection approach. I found one page describing some of it (here)[https://medium.com/python-features/how-to-avoid-classic-pitfall-while-passing-default-values-in-python-7002c0dc4c7c]. It's more important for objects, but it's a protection approach here because `None` is an invalid, but acceptable input. I can go ahead and remove it for primitive values if you'd like.

Reply:
Yeah, default object types are a for sure no-no in python, but "copy" types in python are totally fair to use as defaults and I think make the code more clear. If you feel obliged to change it, go for it! But im not going to hold this up over something this small!